src/lib/components/Swap/SwapButton.tsx

```diff
@@ -129,18 +131,46 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
 
   //@TODO(ianlapham): add a loading state, process errors
   const setDisplayTxHash = useUpdateAtom(displayTxHashAtom)
+
   const onConfirm = useCallback(() => {
     swapCallback?.()
-      .then((transactionResponse) => {
-        // TODO(ianlapham): Add the swap tx to transactionsAtom
-        console.log(transactionResponse)
-        setDisplayTxHash(transactionResponse.hash)
+      .then((response) => {
+        setDisplayTxHash(response.hash)
+        if (inputCurrency && outputCurrency && inputCurrencyAmount && outputCurrencyAmount) {
```

Comment:
Will this always be true? If so, use 'invariant' from 'tiny-invariant' instead of an if statement.

Reply:
ah was wondering why that was used - yeah it always will be